src/aind_codeocean_utils/models/config.py

```diff
@@ -0,0 +1,161 @@
+    )
+    public: bool = Field(
+        default=False, description="Whether the data asset is public or not."
+    )
+    keep_on_external_storage: bool = Field(
+        default=True,
+        description="Whether to keep the data asset on external storage.",
+    )
+    tags: Optional[List[str]] = Field(
+        default=None, description="The tags to use to describe the data asset."
+    )
+    custom_metadata: Optional[Dict] = Field(
+        default=None,
+        description="What key:value metadata tags to apply to the asset.",
+    )
+    viewable_to_everyone: bool = Field(
+        default=False,
+        description="Whether to share the captured results with everyone.",
+    )
+
+    @model_validator(mode="after")
+    def check_tags_and_custom_metadata(self) -> "RegisterDataConfig":
+        """Check that tags and custom_metadata are lists and dicts"""
+        if self.tags is None:
+            self.tags = []
+        if self.custom_metadata is None:
+            self.custom_metadata = {}
+        return self
+
+
+class RunCapsuleConfig(BaseModel):
+    """
+    Settings for running a capsule
+    """
+
+    capsule_id: Optional[str] = Field(
+        default=None, description="ID of the capsule to run."
+    )
+    pipeline_id: Optional[str] = Field(
+        default=None, description="ID of the pipeline to run."
+    )
+    data_assets: Optional[List[Dict]] = Field(
+        default=None,
+        description=(
+            "List of data assets for the capsule to run against. "
+            "The dict should have the keys id and mount."
+        ),
+    )
+    run_parameters: Optional[List] = Field(
+        default=None, description="The parameters to pass to the capsule."
+    )
+    pause_interval: Optional[int] = Field(
+        default=300,
+        description="How often to check if the capsule run is finished.",
+    )
+    capsule_version: Optional[int] = Field(
+        default=None,
+        description="Run a specific version of the capsule to be run.",
+    )
+    timeout_seconds: Optional[int] = Field(
+        default=None,
+        description=(
+            "If pause_interval is set, the max wait time to check if the "
+            "capsule is finished."
+        ),
+    )
+
+    @model_validator(mode="after")
+    def check_data_assets(self) -> "RunCapsuleConfig":
+        """
+        Check that data_assets is a list of dicts with keys 'id' and 'mount'.
+        """
+        if self.data_assets is not None:
+            assert isinstance(
+                self.data_assets, (list, tuple)
+            ), "data_assets must be a list or tuple"
+            assert all(
+                [
+                    "id" in data_asset and "mount" in data_asset
+                    for data_asset in self.data_assets
+                ]
+            ), "data_assets must be a list of dicts with keys 'id' and 'mount'"
+        else:
+            self.data_assets = []
```

Comment:
This is the schema provided by the Code Ocean Swagger docs. We probably don't need to map `None -> []`
```
class CreateDataAssetRequest(BasicRequest):
    name: str
    tags: List[str]
    mount: str
    description: Optional[str] = None
    source: Optional[Source] = None
    target: Optional[Target] = None
    custom_metadata: Optional[Dict[str, Any]] = None


class RunCapsuleRequest(BasicRequest):
    capsule_id: Optional[str] = None
    pipeline_id: Optional[str] = None
    version: Optional[int] = None
    resume_run_id: Optional[str] = None
    data_assets: Optional[List[ComputationDataAsset]] = None
    parameters: Optional[List[str]] = None
    named_parameters: Optional[List[ComputationNamedParameter]] = None
    processes: Optional[List[ComputationProcess]] = None
```